src/tracers.jl

```diff
@@ -191,12 +191,12 @@ HessianTracer(t::HessianTracer) = t
 """
 $(TYPEDEF)
 
-Dual number type keeping track of the results of a primal computation as well as a tracer.
+Dual `Real` number type keeping track of the results of a primal computation as well as a tracer.
 
 ## Fields
 $(TYPEDFIELDS)
 """
-struct Dual{P<:Number,T<:Union{ConnectivityTracer,GradientTracer,HessianTracer}} <:
+struct Dual{P<:Real,T<:Union{ConnectivityTracer,GradientTracer,HessianTracer}} <:
```

Comment:
This is all we need.

Reply:
I thought so too. But then I ran the tests and found that leaving the `::Number` dispatches led to some method ambiguities, typically between
```julia
<=(::Real, ::Real)
<=(::SCT.Dual, ::Number)
```

Reply:
I would even argue that it would be more informative to have `MethodError`s due to type restrictions occur here rather than in `Base.promote_rule`.

Reply:
See the discussion below, we shouldn't promise complex number support anyway. Tracers are real numbers, and if people want to go beyond they can do `Complex{Tracer}`